### PR TITLE
EDU-396: Improve Retry Policy Wording

### PR DESCRIPTION
**Pull request by DJSanti**
## What does this PR do?
This PR adds a correction to non-retryable errors.

### CODE REVIEW

File: docs-src/concepts/what-is-a-retry-policy.md [hunk 1]
```diff
@@ -91,8 +91,8 @@ Non-Retryable Errors = []
 
 - **Description**: Specifies errors that shouldn't be retried.
   - **Default is none.**
-  - If one of those errors occurs, the [Activity Task Execution](#activity-task-execution) or [Workflow Execution](#workflow-execution) is not retried.
-  - The errors are matched against the `type` field of the [ApplicationFailure](/kb/failures#application-failure).
+    - The errors are matched against the `type` field of the [ApplicationFailure](/kb/failures#application-failure).
+  - If one of those errors occurs, the [Workflow Execution](#workflow-execution) won't be retried unless Workflow Retry options are set to allow retries. [Activity Task Executions](#activity-task-execution) will retry unless `temporal.NewNonRetyableError` is returned from the Activity.
```
Comment (cretz) [inline docs-src/concepts/what-is-a-retry-policy.md:95]: Some problems here:

* This is a bit confusing. If one of those errors occurs, the thing won't be retried, regardless of whether that "thing" is workflow execution or activity task execution.
* Throwing/returning/raising an application error that is marked non-retryable won't retry the thing, regardless of whether that "thing" is workflow execution or activity task execution.
* `temporal.NewNonRetyableError` (sic, fix spelling) is a Go only thing but this is general documentation

File: docs-src/concepts/what-is-a-retry-policy.md [hunk 2]
```diff
@@ -91,8 +91,8 @@ Non-Retryable Errors = []
 
 - **Description**: Specifies errors that shouldn't be retried.
   - **Default is none.**
-  - If one of those errors occurs, the [Activity Task Execution](#activity-task-execution) or [Workflow Execution](#workflow-execution) is not retried.
-  - The errors are matched against the `type` field of the [ApplicationFailure](/kb/failures#application-failure).
+    - The errors are matched against the `type` field of the [ApplicationFailure](/kb/failures#application-failure).
+  - If one of those errors occurs, neither the [Workflow Execution](#workflow-execution) nor the [Activity Task Executions](#activity-task-execution) will be retried.
```
Comment (cretz) [inline docs-src/concepts/what-is-a-retry-policy.md:95]:
```suggestion
  - If the error is matched, a retry will not occur.
```

I think that's simpler.

**Review by cretz — APPROVED** [review]
LGTM. Suggestion for simpler wording, but not blocking.

**Review by cretz — APPROVED** [review]

File: docs-src/concepts/what-is-a-retry-policy.md [hunk 3]
```diff
@@ -91,8 +91,8 @@ Non-Retryable Errors = []
 
 - **Description**: Specifies errors that shouldn't be retried.
   - **Default is none.**
-  - If one of those errors occurs, the [Activity Task Execution](#activity-task-execution) or [Workflow Execution](#workflow-execution) is not retried.
-  - The errors are matched against the `type` field of the [ApplicationFailure](/kb/failures#application-failure).
+    - The errors are matched against the `type` field of the [ApplicationFailure](/kb/failures#application-failure).
+  - If one of those errors occurs, a retry will not occur.
```
Comment (djmagee) [inline docs-src/concepts/what-is-a-retry-policy.md:95]:
```suggestion
  - Errors are matched against the `type` field of the [Application Failure](/kb/failures#application-failure).
  - If one of those errors occurs, a retry does not occur.
```
Recommend adding a space to "ApplicationFailure" to account for the varying names amongst the SDKs.

File: docs-src/concepts/what-is-a-retry-policy.md [hunk 4]
```diff
@@ -91,8 +91,8 @@ Non-Retryable Errors = []
...
 - **Use case**: There may be errors that you know of that should not trigger a retry.
   In this case, you can specify them such that, if they occur, the given execution is not retried.
```
Comment (djmagee) [inline docs-src/concepts/what-is-a-retry-policy.md:97]:
```suggestion
- **Use case**: If you know of errors that should not trigger a retry, you can specify that, if they occur, the execution is not retried.
```
Technically out of scope for this PR, but conveniently within my reach.

**Review by djmagee — DISMISSED** [review]

**Review by djmagee — APPROVED** [review]

**Pull request merged**